### ISSUE TITLE
U+002D HYPHEN-MINUS in <mo> operators

### ISSUE THREAD

**Issue by fred-wang** [w3c/mathml-core#70]
U+002D HYPHEN-MINUS is too short, so MathML browser implementations render it as U+2212 MINUS SIGN.

Do we want to standardize this workaround?

I'd prefer not, but I guess the proper way to do it would be via a new text-transform value for mo.

If not, tools should generate the proper code point instead and we can write a polyfill for that.

**Comment by MurraySargent** [comment]
Most definitely replace U+002D by U+2212 in a math zone (inside <math>…</math>) unless it’s explicitly marked as <mtext>. No other if ands or buts 😊 Seriously, U+002D looks simply awful as a minus sign. That’s why we encoded U+2212 for the minus sign.

Thanks,
Murray

From: Frédéric Wang ***@***.***>
Sent: Monday, June 28, 2021 7:50 PM
To: w3c/mathml-core ***@***.***>
Cc: Murray Sargent ***@***.***>; Comment ***@***.***>
Subject: [w3c/mathml-core] U+002D HYPHEN-MINUS in <mo> operators (#70)


U+002D HYPHEN-MINUS is too short, so MathML browser implementations render it as U+2212 MINUS SIGN.

Do we want to standardize this workaround?

I'd prefer not, but I guess the proper way to do it would be via a new text-transform value for mo.

If not, tools should generate the proper code point instead and we can write a polyfill for that.

**Comment by fred-wang** [comment]
I think there are confusions in the this thread. Editors are free (and probably should) to replace any typed U+002D (-) with U+2212 (−), that's out of the scope of MathML Core. The question is about whether we want to introduce a hack (e.g. based on text-transform) in browsers with all the extra issues it opens (more exceptions in the code, more tests neeed, text mistatch between DOM / rendered / ATs / copy & paste, etc).

In any case, there is no plan to integrate such a hack in Chromium's initial implementation so I guess this should be level 2.

**Comment by NSoiffer** [comment]
There is no confusion in my mind -- this is a requirement for the spec, not editors. Existing MathML and existing MathML producers mostly use U+002D and expect it to render with the U+2212 glyph.

Here's the difference illustrated with a trivial [codepen](https://codepen.io/nms/pen/BaweOMB) rendered by Chrome with math support:
![image](https://user-images.githubusercontent.com/1545836/150278218-241ad87f-0ade-4bf6-b7c0-ff71c2da04aa.png)

I don't think the spec should say how to implement this equivalence. It should merely say that  U+002D should be rendered as U+2212. I'm not convinced that text-transform is the only way this can be done. What I am convinced is that not implementing this equivalence is a significant change from people's expectation and current use.

**Comment by faceless2** [comment]
I completely agree with Neil. At the absolute least you need some sort of statement of intent - _it is expected that user-agents will convert U+002D to U+2212 for both rendering and the AT tree via some undefined mechanism_, even if it's marked as optional, would go some way to limiting the inevitable divergence when implementers are given existing MathML docs, and the existing MathML users that go with them, and a spec that's silent on what to do about it.

(I say inevitable with some confidence, because we've implemented this substitution).

**Comment by davidcarlisle** [comment]
I think (whether or not you can implement this in the first implementation) that core should say that - should render as minus, the majority of existing MathML assumes this, including all instances of subtraction in the MathML3 spec and almost all existing generators eg tex to mathml convertors. I don't think you can brush this off as "confusion" on the part of the commenters or that it is a "hack". The - symbol in Unicode is explicitly dual use HYPHEN-MINUS and should act as a hyphen in text and a minus in math.

**Comment by bkardell** [comment]
Discussed in the meeting today, no resolutions. We'll circle back on this next month after we have some more async discussions.

**Comment by NSoiffer** [comment]
Adding needs-spec change label as a meeting agreement was this must be done due to the vast amount of legacy MathML that assumes this equivalence. Both Firefox and Safari support this. If Chrome can't handle this, it should fail a test.

**Comment by bkardell** [comment]
Assigning to myself not for the edits but to sort out what we're going to do here/proposals and to write tests.

**Comment by davidcarlisle** [comment]
I will write a PR for the mathml-core spec. Investigations ongoing on whether this can be implemented.
